test/network_proposal_test.rb

```diff
@@ -0,0 +1,64 @@
+require_relative "test_helper"
+
+require "network/clients/network_proposal"
+
+class DummyLanItems
+  def summary(params)
+    params == "one_line" ? "one line summary" : "rich_text_summary"
+  end
+end
+
+class DummyLan
+  def Summary(_)
+    ["rich_text_summary"]
+  end
+end
+
+describe Yast::NetworkProposal do
+  subject { Yast::NetworkProposal.new }
+
+  let(:lan_items_mock) { DummyLanItems.new }
+  let(:lan_mock) { DummyLan.new }
+
+  before do
+    stub_const("Yast::LanItems", lan_items_mock)
+    stub_const("Yast::Lan", lan_mock)
+  end
+
+  describe "#description" do
+    it "returns a map with id, menu_title and rich_text_title " do
+      expect(subject.description).to include("id", "menu_title", "rich_text_title")
+    end
+  end
+
+  describe "#make_proposal" do
+    it "returns a map with 'label_proposal' as an array with one line summary'" do
+      expect(subject.make_proposal({})["label_proposal"]).to eql(["one line summary"])
+    end
+
+    it "returns a map with 'preformatted_proposal' as an array with the network summary'" do
+      expect(subject.make_proposal({})["preformatted_proposal"]).to eql(["rich_text_summary"])
+    end
+  end
+
+  describe "#ask_user" do
+    Yast.import "Wizard"
+
+    before do
+      allow(Yast::Wizard)
```

Comment:
Just thinking in allow any Wizard method to not mock the open and close dialog, so used null object at the end.